profiles.wordpress.org/public_html/wp-content/plugins/wporg-profiles-activity-handler/wporg-profiles-activity-handler.php

```diff
@@ -234,6 +253,10 @@ public function handle_activity() {
 					throw new Exception( '-1 Unable to save activity' );
 				}
 
+				if( $activity instanceof SavedActivity && $activity->is_contribution ) {
```

Comment:
The `SavedActivity` class could also be used for all the code paths in the `handle_` functions by creating other properties for the other returns types. Ie ($activity->error_msg, etc...)